android/src/main/java/com/swmansion/reanimated/keyboard/Keyboard.java

```diff
@@ -20,15 +23,30 @@ public int getHeight() {
   public void updateHeight(WindowInsetsCompat insets) {
     int contentBottomInset = insets.getInsets(contentTypeMask).bottom;
     int systemBarBottomInset = insets.getInsets(systemBarTypeMask).bottom;
-    int keyboardHeightDip = contentBottomInset - systemBarBottomInset;
-    height = (int) PixelUtil.toDIPFromPixel(Math.max(0, keyboardHeightDip));
+    boolean hasNavigationBar = KeyCharacterMap.deviceHasKey(KeyEvent.KEYCODE_HOME);
+    int keyboardHeightDip =
+        hasNavigationBar ? contentBottomInset - systemBarBottomInset : contentBottomInset;
+    int keyboardHeight = (int) PixelUtil.toDIPFromPixel(Math.max(0, keyboardHeightDip));
+    if (keyboardHeight == 0 && state == KeyboardState.OPEN) {
+      return;
+    }
+    height = keyboardHeight;
   }
 
   public void onAnimationStart() {
-    state = height == 0 ? KeyboardState.OPENING : KeyboardState.CLOSING;
+    if (activeTransitionCounter > 0) {
```

Comment:
Please check if when switching between two text inputs rapidly you won't get this even re-triggered even though the keyboard shouldn't close.

Reply:
I've checked this and it works 🎉